### PR TITLE
Update documentation for the package

### PR DESCRIPTION
**Pull request by hskrasek**
As the title says, working on updating the documentation for the package to try and bring it in line with the other packages.

Signed-off-by: Hunter Skrasek <hunterskrasek@me.com>

### CODE REVIEW

**Comment by hskrasek** [comment]
Currently a work in progress, just wanted to open early so that people can catch any spelling mistakes or weird grammar screw ups.

**Comment by hskrasek** [comment]
@kelunik I'm down with that. I was mainly just mirroring how the aerys package has its classes documented. Do you think we should document the `Connection` and `Parser` classes? `BeanstalkClient` will get documented once I finish adding examples for the other methods.

**Comment by hskrasek** [comment]
Fyi, still gonna finish this, just got tied up the past couple of days. Soon though! Soon

**Comment by hskrasek** [comment]
@kelunik If I wanted to see how these docs look locally, whats the best way? I got Jeykll running locally but if I start it up within the docs directory everything just comes out unstyled.

**Comment by kelunik** [comment]
@hskrasek See https://github.com/amphp/amphp.github.io#setup-for-an-existing-repository

**Comment by hskrasek** [comment]
@kelunik Sorted out the integer vs int issue. Sorry this is taking so long, wanna add a couple of pages about working with jobs and tubes. Haven't directly worked with some of the commands though, so been studying up on the protocol.

Might actually throw together some integration tests (and submit those separately) to make sure my examples make sense. Hope to have this finished up for you soon though

**Comment by hskrasek** [comment]
Alright I'm sure some wording needs to be adjusted all over the place, but I think I've got just about everything documented with examples.

File: docs/jobs.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Getting a jobs stats
+
+```php
+$beanstalk = new Amp\Beanstalk\BeanstalkClient("tcp://127.0.0.1:11300");
+
+// This step not required if you included a tube query parameter when creating the client
+$beanstalk->use('foobar');
+
+$jobStats = $beanstalk->getJobStats($jobId = 42);
```
Comment (kelunik) [inline docs/jobs.md]: Missing yield

File: docs/misc.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{:toc}
+
+## Get system stats
+
+To see what stats are available for the system, checkout the [System](classes/system) class page.
+
+```php
+$beanstalk = new Amp\Beanstalk\BeanstalkClient("tcp://127.0.0.1:11300");
+
+$stats = $beanstalk->getSystemStats();
```
Comment (kelunik) [inline docs/misc.md]: Missing yield

File: docs/tubes.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+title: Working with Tubes
+permalink: /tubes
+---
+
+* Table of Contents
+{:toc}
+
+## Using a different tube
+
+By default Beanstalk will use the default tube for reserving and storing new jobs. To work with a different tube, you need can use `use`:
```
Comment (kelunik) [inline docs/tubes.md]: Either need or can, but not both

File: docs/tubes.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+## Pausing a tube
+
+If you need to pause a tube, preventing any new jobs from being reserved, you can do the following:
+
+```php
+$beanstalk = new Amp\Beanstalk\BeanstalkClient("tcp://127.0.0.1:11300");
+
+$beanstalk->pause($tube = 'foobar');
```
Comment (kelunik) [inline docs/tubes.md]: Missing yield

File: docs/tubes.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+By default when you reserve a job you'll either pull from the `default` tube, or the tube you `use`ed. If you'd like to reserve jobs from other tubes, you can use `watch` to pull from multiple tubes. If you need to remove a job from the watch list, you can use `ignore`.
+
+```php
+$beanstalk = new Amp\Beanstalk\BeanstalkClient("tcp://127.0.0.1:11300");
+
+$beanstalk->watch($tube = 'foobar');
+$beanstalk->watch($tube = 'barbaz');
+$beanstalk->ignore($tube = 'default');
+// Watchlist will contain "foobar" and "barbaz"
+```
```
Comment (kelunik) [inline docs/tubes.md]: Missing yields

**Comment by kelunik** [comment]
Should I merge this now or do you plan further changes?

**Comment by hskrasek** [comment]
@kelunik Hmm I can't think of anything else to add at the moment. Noticed some missing methods from the client that exist in the beanstalk spec, but I'll document those when I get around to PRing them (unless someone else beats me to it).

I think we're all good here

**Comment by kelunik** [comment]
@hskrasek Could you open issues for the missing methods?

Thanks for your contribution!

**Pull request merged**